cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -69,6 +71,48 @@ struct dispatch_void_if_nested {
 };
 
 namespace row {
+
+enum class lhs_index_type : size_type {};
+enum class rhs_index_type : size_type {};
+
+template <typename Index, typename Underlying = std::underlying_type_t<Index>>
+struct strong_index_iterator : public thrust::iterator_facade<strong_index_iterator<Index>,
```

Comment:
@bdice I realize that I am late to the party here, but this struct has zero docs. A new reader of the code won't have any idea _why_ this struct exists since the purpose is not conveyed by its implementation, but rather by its _existence_ alone. Could you make a PR with some brief documentation (or maybe just stick that into one of the downstream PRs that either you or @ttnghia is working on)?

Reply:
Sure thing. I can create a standalone PR for this.

Reply:
Thank you!

Reply:
https://github.com/rapidsai/cudf/pull/10888